mce-capi-webhook-config/webhook/webhook.go

```diff
@@ -0,0 +1,151 @@
+package hook
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/go-logr/logr"
+	"gomodules.xyz/jsonpatch/v2"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kube-openapi/pkg/validation/errors"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var log = logf.Log.WithName("example-controller")
+
+// +kubebuilder:webhook:path=/mutate,mutating=true,failurePolicy=fail,groups="cluster.x-k8s.io",verbs=create;update,versions=v1,name=mce-capi-webhook-config.x-k8s.io
+
+// MceCapiWebhookConfig label MCE objects for groups="cluster.x-k8s.io"
+type MceCapiWebhookConfig struct {
+	Client         client.Client
+	MceLabelConfig *Config
+	ClientSet      *kubernetes.Clientset
+	Log            logr.Logger
+}
+
+type Config struct {
+	NamespaceOpenshiftClusterApi string
+	HyperShiftLabelName          string
+	LabelMultiClusterEngine      string
+}
+
+func NewConfig() *Config {
+	return &Config{
+		NamespaceOpenshiftClusterApi: "openshift-cluster-api",
+		HyperShiftLabelName:          "hypershift.openshift.io/hosted-control-plane",
+		LabelMultiClusterEngine:      "multicluster-engine",
+	}
+}
+
+func (li *MceCapiWebhookConfig) countLabel(ctx context.Context, namespaceName string, oldValue string) (string, bool, error) {
+	/*
+		* The auto-labeling mutating webhook inspects the NS
+		  * If the namespace is openshift-cluster-api, don't label for MCE
+		  * Else, fetch the NS and inspect labels
+		    * If it has the hypershift label, don't label for MCE
+		  * If the NS is for (HyperShift or openshift-cluster-api) AND already has the MCE label, reject the admission, invalid configuration
+		  * If we've not returned by now, add the MCE label
+		* Relevant controller acts on CR
+		  * MCE using a watchfilter on their label
+	*/
+
+	newLabel := ""
+
+	isValidNamespace := true
+	// If the namespace is openshift-cluster-api, don't label for MCE
+	if namespaceName == li.MceLabelConfig.NamespaceOpenshiftClusterApi {
+		isValidNamespace = false
+	} else {
+		// Else, fetch the NS and inspect labels
+		namespace, err := li.ClientSet.CoreV1().Namespaces().Get(ctx, namespaceName, metav1.GetOptions{})
+		if err != nil {
+			return "", false, errors.New(0, "Cannot get namespace='%s' : %s", namespaceName, err.Error())
+		}
+		hyperShiftValue, isHyperShift := namespace.Labels[li.MceLabelConfig.HyperShiftLabelName]
+		if isHyperShift && hyperShiftValue == "true" {
+			isValidNamespace = false
+		}
+	}
+
+	// If the NS is for (HyperShift or openshift-cluster-api) AND already has the MCE label, reject the admission, invalid configuration
+	if !isValidNamespace {
+		if oldValue == li.MceLabelConfig.LabelMultiClusterEngine {
+			return "", false, errors.New(0, "Invalid configuration, cannot use label %s", oldValue)
```

Comment:
```suggestion
			return "", false, errors.New(0, "Invalid configuration, cannot use label %q", oldValue)
```

`%q` puts a string in escaped quotes. I find it can be helpful when reading error messages, but this is just a suggestion, not a blocker by any means.